File: Data Structures & Algorithms/validate-parentheses/submission-0.cpp
class Solution {
public:
    bool isValid(string s) {
        stack <char>st;
        int cnt = 0 ;
        for (int i =0 ;i < s.size();i++) {
            if (s[i] == '(' || s[i] == '{' || s[i] == '[') {
                st.push(s[i]);
            }else {
                cnt++;
                if (!st.empty()) {
                    if (s[i] == ')' && st.top() == '(')st.pop(),cnt--;
                    else if (s[i] == '}' && st.top() == '{')st.pop(),cnt--;
                    else if (s[i] == ']' && st.top() == '[')st.pop(),cnt--;
                }
            }
        }
        return !cnt && st.empty();
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        ans = [0] * len(temperatures)
        stack = []

        for idx in range(len(temperatures) - 1, -1, -1):
            while stack and temperatures[stack[-1]] <= temperatures[idx]:
                stack.pop()

            ans[idx] = stack[-1] - idx if stack else 0
            stack.append(idx)

        return ans